wsd/DocumentBroker.cpp

```diff
@@ -1491,9 +1491,33 @@ void DocumentBroker::handleUploadToStorageResponse(const StorageBase::UploadResu
             Poco::URI::encode(filename, "", encodedName);
             const std::string filenameAnonym = LOOLWSD::anonymizeUrl(filename);
 
+            constexpr const char* needle = "?access_token";
+            std::size_t index = getPublicUri().toString().find(needle);
+            std::string url1, url2;
+            if (index != std::string::npos)
+            {
+
+                url1 = getPublicUri().toString().substr(0, index);
+                url2 = url.substr(0, index);
+            }
+            else
+            {
+                url1 = url;
+                url2 = url;
+            }
+
             std::ostringstream oss;
             oss << "renamefile: " << "filename=" << encodedName << " url=" << url;
             broadcastMessage(oss.str());
+            if (!strcasecmp(url1.c_str(), url2.c_str()))
+            {
+                // WOPISrc has not changed
+                // However, due to rename, timestamp in the storage has changed
+                // and any further upload requests will result to a conflict
+                // Do notify the client about the conflict
+                // This will make the client reload the document properly
```

Comment:
I wonder why couldn't we do this in the JS code? We do send `renamefile:` with the new URL, right? And it has the old one. So, it knows we are renaming and has the old and new url, why can't it close and reopen? Why should WSD send it an explicit `close:`?